### DISCUSSION TITLE
Way to view back end of drivebase functions?

### DISCUSSION THREAD

**Discussion by Mathi256** [pybricks/support#1718] [Q&A] [answered]
Hello, I was searching for a way to view the code for some drivebase functions like drivebase.straight or drivebase.turn. I would like to be able to view them since I would like to make my own modified version of both. Is there a way to view the programs of those functions?

**Comment by dlech** [comment]
Sure, you can find the drivebase code here: https://github.com/pybricks/pybricks-micropython/blob/master/lib/pbio/src/drivebase.c and the Python wrapper code here: https://github.com/pybricks/pybricks-micropython/blob/master/pybricks/robotics/pb_type_drivebase.c

In most cases though, you can just tweak the settings to suit your needs instead of modifying the firmware.

**Comment by laurensvalk** [comment, marked as answer]
What would you like to change?

> I would like to make my own modified version of both

Instead of modifying the base class in the firmware, Python lets you make your own derived class as follows.

Suppose for example that you wanted to change the `straight` or `turn` methods to print a message to say how far they will move:

```python
from pybricks.parameters import Direction, Port, Stop
from pybricks.pupdevices import Motor
from pybricks.robotics import DriveBase
from pybricks.tools import wait


class MyOwnDriveBase(DriveBase):

    def turn(self, angle, then=Stop.HOLD, wait=True):
        print("I'm turning by", angle, "degrees")
        super().turn(angle, then, wait)

    def straight(self, distance, then=Stop.HOLD, wait=True):
        print("I'm driving straight by", distance, "mm")
        super().straight(distance, then, wait)


left = Motor(Port.A, Direction.COUNTERCLOCKWISE)
right = Motor(Port.B, Direction.CLOCKWISE)
robot = MyOwnDriveBase(left, right, 56, 128)


robot.straight(100)
robot.turn(90)
```

**Reply by Mathi256** [reply]
Thank you for the info, I’m going to use this instead!
